ang/civicase/MasonryGrid.js

```diff
@@ -0,0 +1,118 @@
+(function (_, $, angular) {
+  var module = angular.module('civicase');
+
+  module.directive('civicaseMasonryGrid', function ($timeout) {
+    return {
+      controller: 'civicaseMasonryGridController',
+      link: civicaseMasonryGridLink
+    };
+
+    /**
+     * Masonry Grid link function.
+     *
+     * @param {Object} $scope the directive's scope.
+     * @param {Object} $element a reference to the directive's element.
+     * @param {Object} attrs a map of attributes associated to the element.
+     * @param {Object} masonryGrid a reference to the directive's controller.
+     */
+    function civicaseMasonryGridLink ($scope, $element, attrs, masonryGrid) {
+      (function init () {
+        $timeout(function () {
+          appendMasonryContainers();
+          arrangeGridItems();
+          $scope.$on('civicaseMasonryGrid::updated', arrangeGridItems);
+        });
+      })();
+
+      /**
+       * Appends the masonry containers used for splitting the grid items.
+       */
+      function appendMasonryContainers () {
+        _.times(2).forEach(function () {
+          $('<div></div>')
+            .addClass('civicase__summary-tab-tile')
+            .addClass('civicase__masonry-grid__container')
+            .appendTo($element);
+        });
+      }
+
+      /**
+       * Sorts the grid items into one of the two grid containers.
+       */
+      function arrangeGridItems () {
+        $timeout(function () {
+          $element.find('[civicase-masonry-grid-item]').detach();
+
+          masonryGrid.$gridItems.forEach(function ($gridItem, index) {
+            var columnIndex = index % 2;
+            var $column = $element.find('.civicase__masonry-grid__container').eq(columnIndex);
+
+            $gridItem.appendTo($column);
+          });
+        });
+      }
+    }
+  });
+
+  module.controller('civicaseMasonryGridController', function ($scope) {
+    var vm = this;
+    vm.$gridItems = [];
+
+    /**
+     * Adds an item element to the grid.
+     *
+     * @param {Object} $gridItem a reference to the item element to be added.
+     */
+    vm.addGridItem = function ($gridItem) {
+      _.remove(vm.$gridItems, $gridItem);
```

Comment:
you are doing this in basically every method, please put it in a separate function and document it

Reply:
@reneolivo you didn't reply nor act on this comment